backend/src/main/java/net/es/oscars/web/rest/MapController.java

```diff
@@ -64,13 +64,14 @@ public MapGraph getMap() throws ConsistencyException, StartupException {
                     .type(d.getType().toString())
                     .build();
 
-            if (positionMap.keySet().contains(d.getUrn())) {
-                n.setX(positionMap.get(d.getUrn()).getX());
-                n.setY(positionMap.get(d.getUrn()).getY());
-                n.setFixed(new HashMap<>());
-                n.getFixed().put("x", true);
-                n.getFixed().put("y", true);
-
+            for (String key: positionMap.keySet()) {
```

Comment:
do some checking here in case you get multiple matches though.

Reply:
> @haniotak I'll add the missed condition. This above condition already checks if a device is present more than once in the positionMap i.e it checks the case for multiple positions

Condider what happens if your device is `bois-cr5` and your position keys are `bois` and `boi`.